interpreter/cling/lib/Interpreter/DynamicLibraryManager.cpp

```diff
@@ -152,12 +187,36 @@ namespace cling {
         return std::string();
     }
 
-    std::string foundName = lookupLibMaybeAddExt(libStem);
-    if (foundName.empty() && !libStem.startswith("lib")) {
-      // try with "lib" prefix:
-      foundName = lookupLibMaybeAddExt("lib" + libStem.str());
+    // Handle substitutions
+    std::string result;
+    llvm::StringRef mainExecutable =
+        llvm::sys::fs::getMainExecutable(nullptr, nullptr);
+    if (libStem.startswith_lower("@rpath/")) {
+      llvm::SmallVector<llvm::StringRef, 2> RPaths;
+      SplitPaths(RPATH, RPaths, utils::kPruneNonExistant, platform::kEnvDelim,
```

Comment:
Pre-existing spell-o:
```suggestion
      SplitPaths(RPATH, RPaths, utils::kPruneNonExistent, platform::kEnvDelim,
```

Reply:
Yes please :-)